### PR TITLE
Text Intent Suite

### LINKED ISSUE #38: Create a TextIntent
#### Description

Before we start to apply Foreshadow on open dataset for benchmarking, we need to add a TextIntent to apply basic TF-IDF transformation.

### PR DESCRIPTION
**Pull request by adithyabsk**
WIP SmartText, HTMLRemover, and related tests. Fix pytest.ini file. Update poetry.lock and pyproject.toml to include lxml. Broken tests because TfidfVectorizer cannot be wrapped

### Related Issue
Fixes #38

### CODE REVIEW

File: foreshadow/intents/general.py [hunk 1]
```diff
@@ -191,8 +200,41 @@ def column_summary(cls, df):
...
+        """Returns standard computed statistics for a TextIntent column
+
+            The following are computed:
+                nan: count of nans pass into dataset
+                mode: mode or np.nan if data is mostly unique
+                top10: top 10 most frequent values or empty array if mostly unique
+                    [(value, count),...,]
+        """
+
+        return _standard_col_summary(df)
```
Comment (robocoder99) [inline foreshadow/intents/general.py]: Is this summary adequate for text-based columns? I don't have a lot of experience with ngrams and text vectorizers but it seems there may be more useful diagnostics to get.

File: foreshadow/transformers/smart.py [hunk 2]
```diff
@@ -217,3 +220,49 @@ def _get_transformer(self, X, y=None, **fit_params):
...
+        if (data.dtype.type is not np.str_) and not all(
+            [isinstance(i, str) for i in data]
+        ):
+            steps.append(("num", ToString()))
+
+        html_ratio = (data.astype("str").apply(HTMLRemover.is_html).sum()) / len(data)
+        if html_ratio > self.html_cutoff:
+            steps.append(("hr", HTMLRemover()))
+
+        # TODO: find heuristic for finding optimal values for values
```
Comment (robocoder99) [inline foreshadow/transformers/smart.py]: Do we need to create an issue for this?

File: foreshadow/tests/test_foreshadow.py [hunk 3]
```diff
@@ -333,37 +333,53 @@ def test_foreshadow_predict_diff_cols():
...
+        estimator=DummyRegressor(),
+        optimizer=DummySearch,
+    )
+
+    fs2.fit(x, y)
+    assert isinstance(fs2.pipeline.steps[-1][1], DummyRegressor)
 
 
-def test_foreshadow_param_optimize():
+def test_foreshadow_param_optimize():  # TODO: Make this test faster
```
Comment (robocoder99) [inline foreshadow/tests/test_foreshadow.py]: Do we need an issue from this?

**Comment by robocoder99** [comment]
Its also worth noting that the dev branch has diverged significantly from this one since it was forked. You might want to merge in from dev before attempting a merge in the other direction. Thats a matter of personal opinion / advice though.

File: foreshadow/intents/general.py (hunk 1 above)
Comment (adithyabsk) [reply foreshadow/intents/general.py]: I think so, I looked around and did not note any other summary statistics that might be useful for text based columns. @robocoder99 or @jichaogp if you think of any others let me know.

File: foreshadow/tests/test_foreshadow.py (hunk 3 above)
Comment (adithyabsk) [reply foreshadow/tests/test_foreshadow.py]: Will do

File: foreshadow/transformers/smart.py (hunk 2 above)
Comment (adithyabsk) [reply foreshadow/transformers/smart.py]: #55 Done

File: foreshadow/tests/test_foreshadow.py (hunk 3 above)
Comment (adithyabsk) [reply foreshadow/tests/test_foreshadow.py]: #56 Done.

**Comment by adithyabsk** [comment]
Thanks for the feedback @robocoder99, addressed your comments.

**Pull request merged**